### PR TITLE
apt: add support for package version >=

### LINKED ISSUE #69034: Apt module to support minimum version specification ">="
It's a reopening of the closed issue #21449.

##### SUMMARY

I would like to be able to specify the minimal version of a package to be installed instead of a (partial) exact match.

For example, I would like to have elasticsearch present on my system in a version equal or greater to 7.6.1 because feature X is present since this version:
- if elasticsearch is not installed: install the latest elasticsearch package 
- if elasticsearch is installed in version 7.6.0: install the latest elasticsearch package (7.6.2) 
- if elasticsearch is installed in version 7.6.1: do nothing, return ok
- if elasticsearch is installed in version 7.6.2 or more: do nothing, return ok

```yml
    - name: Install Elasticsearch.
      package:
        name: elasticsearch>=7.6.1
        state: present
    
```
##### ACTUAL RESULTS

```
fatal: [poc-elasticsearch]: FAILED! => {"changed": false, "msg": "No package matching 'elasticsearch>' is available"}
```

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME

[apt module](https://docs.ansible.com/ansible/latest/modules/apt_module.html)

##### ADDITIONAL INFORMATION

Answer to  https://github.com/ansible/ansible/issues/21449#issuecomment-280158405 

> Does apt itself support these? If not, I'm not certain we want to add it as that would require us to parse and compare dpkg version strings. If it does, then I could see us taking a PR to use the apt python bindings to do this... Someone would have to reconcile this with the current usage of the "=" to mean fnmatching, though.

Apt python bindings support the comparison of versions with `apt_pkg.version_compare`:

```py 
import apt
import apt_pkg

cache = apt.Cache()
pkg = cache['elasticsearch']

installed = pkg.installed.version
candidate = pkg.versions[0].version

vc = apt_pkg.version_compare(installed,candidate)
if vc > 0:
    print(f'{installed} version > {candidate} version')
elif vc == 0:
    print(f'{installed} version = {candidate} version')
elif vc < 0 and pkg.is_upgradable:
    print(f'{installed} version < {candidate} version')
```
Documentation: https://apt-team.pages.debian.net/python-apt/library/apt_pkg.html?highlight=version_compare#apt_pkg.version_compare

### PR DESCRIPTION
**Pull request by phemmer**
##### SUMMARY

This adds the ability to specify a package version using `>=`. This will ensure the package is at the specified version or above.

* If the package is not installed, the latest version will be installed.
* If the package is installed and less than the specified version, it will be upgraded to the latest version.
* If the package is installed and greater than or equal to the specified version, it will be left alone.

The version selection is handled by Apt itself, so things like the system policy, pinning, etc, are considered.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

apt

##### ADDITIONAL INFORMATION

Closes #69034

### CODE REVIEW

**Comment by bcoca** [comment]
do we just want one set of comparison operators? if we are going down this road I think we should add all of them.

**Comment by phemmer** [comment]
You mean `<=`, `<`, `>`? It can't be done as Apt does not provide this functionality. The module would have to re-implement Apt's policy engine to determine which packages are installation candidates. Unless there are open issues requesting this, I don't think this is a road we want to go down.

File: lib/ansible/modules/apt.py [hunk 1]
```diff
@@ -21,7 +21,7 @@
 options:
   name:
     description:
-      - A list of package names, like C(foo), or package specifier with version, like C(foo=1.0).
+      - A list of package names, like C(foo), or package specifier with version, like C(foo=1.0) or C(foo>=1.0).
```
Comment (Akasurde) [inline lib/ansible/modules/apt.py:24]: Please create a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment). See [this fragment](https://github.com/ansible/ansible/blob/c2f528b7686c9573aa8493357219e0bbe4416dd3/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example. Also, add integration tests for this change.

**Review by Akasurde — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/apt.py (hunk 1 above)
Comment (phemmer) [reply lib/ansible/modules/apt.py:24]: Done

**Comment by lukasjuhrich** [comment]
@Akasurde are there any blockers on this? Your approval state is still „Changes requested“ even though all the comments have been resolved.

This is a really useful feature and not merging this PR due to an oversight would be a shame.

**Comment by Akasurde** [comment]
I am OK with the changes. I am waiting for others to comment.

**Review by bcoca — APPROVED** [review]
I still think we'll end up implementing the other operators

**Comment by evulhotdog** [comment]
@Akasurde what is the number of approvals that you need to merge this?

Is @bcoca's enough?

**Comment by Akasurde** [comment]
> @Akasurde what is the number of approvals that you need to merge this?

2

@bcoca Are you OK with me merging this change? Thanks.

**Review by Akasurde — APPROVED** [review]

**Pull request merged**